lib/engine-addon.js

```diff
@@ -84,10 +84,10 @@ const buildVendorCSSTree = memoize(function buildVendorCSSTree(vendorTree) {
   return this.debugTree(vendorCSSTree, 'vendor-style:input');
 });
 
-const buildEngineJSTree = memoize(function buildEngineJSTree() {
+const buildEngineJSTree = memoize(function buildEngineJSTree(tree) {
   let engineSourceTree;
   let treePath;
-  let addonTree = this.options.trees.addon;
+  let addonTree = tree || this.options.trees.addon;
```

Comment:
Now that `tree` is passed in, under what circumstances should we continue to consult `this.options.trees.addon` ?

My intuition (i'll need to verify) is that your change, removes the need to invoke consult `this.options.trees.addon` at all. Could you investigate?

Reply:
I think that is correct – I kept this fallback in to ensure full back compat. Will test it out a bit more and get back to you.